### ISSUE TITLE
Feasibility question: controllable widget with button

### ISSUE THREAD

**Issue by dommitchell** [DOAJ/doaj#976]
This is just a question.

Re #762 how feasible would it be to provide users with a button as well so, if they made their own search then could also embed a simple button - Find similar articles on DOAJ?

I think this is slightly different to the functionality of the controllable widget but am sure if it is in any way related. Thoughts?

**Comment by Steven-Eardley** [comment]
As it is, the fixed query widget is essentially a container for a set of facetview results. So I can imagine a site having such a button and simply hiding the ```<div>``` until the user wants to see it. I expect this would be feasible in its current state (assuming it doesn't mind being hidden and re-drawn).

Although, I'm not totally sure what you're describing. Maybe you'd like a 3rd sort of widget that takes you to the DOAJ site based on a button and a fixed query, which would be a bit like the simple search widget but without the search box.

**Comment by emanuil-tolev** [comment]
It's one of the main uses we envisioned @dommitchell , except users would have to know how to configure the options to show their articles. We can add a few examples on the widget page to show how to embed DOAJ search results relating to [some phrase], [a subject field of study], [a specific author's exact name]. Is that what you mean?

We kind of have that already via the search page's "embed" functionality - you can embed the results of any search. So, do your search, select your facets (incl. subject), type your phrase in, and hit "embed".

Dynamic setting, like embedding the widget on MDPI's website and having it show "related publications (from this field of study, or from these authors) and have it change dynamically on every MDPI publication page, would require a person with some basic skills in Javascript and access to MDPI's website.

**Comment by emanuil-tolev** [comment]
> type your phrase in,

or an author's name surrounded by double quotes to match the entire name at once.

**Comment by dommitchell** [comment]
Thanks guys. Now I read the user's email again, I think what he was after was a button. A DOAJ button that he could embed on his site that, when clicked, would take people to DOAJ search results. He said:

"The other buttons I use all use a form input process, with the search engine identified as follows:
<form method="get" action="http://www.bing.com" target="_blank">, with an input string for the search terms, so all (!) I need is a url to which the search is directed.  I guess that is the tricky development bit for your team, since it looks as though you use some form of Google site-specific search engine, which uses a whole whack of javascript for its implementation. I tried using a url that would do a google site specific search, which works to some extent, but which also treats doaj.org as a search term!"

**Comment by emanuil-tolev** [comment]
I see, a simple link. How about using the share functionality of the search page?

1\. go to https://doaj.org/search . Perform a search. I performed one for "Isac Karimi" (with the quotes). Copy URL.
2\. put this into html page:

```html
<a href="http://doaj.org/search?source=%7B%22query%22%3A%7B%22query_string%22%3A%7B%22query%22%3A%22%5C%22Isac%20Karimi%5C%22%22%2C%22default_operator%22%3A%22AND%22%7D%7D%2C%22from%22%3A0%2C%22size%22%3A10%7D">
  <button>Search for author Isac Karimi</button>                                                                                                                                        
</a>
```

3\. Replace the href part with the copied URL.
4\. You have a button that goes to the DOAJ search page looking for "Isac Karimi".
5\. Replace ```Isac%20Karimi``` from the middle of that long URL with ```Own%20Name``` (the %20 is a space) to reflect own name to create a customised button.

If *this* is what they want to do, then we could develop a simple tool under T&M to generate such a button - you just punch in your search term, and optionally select a specific field to search (like on the search page UI). It'd probably take about a day, half to develop the thing with a dropdown to select specific fields, and half to write documentation, test and offer basic advice on styling the resulting links/buttons.

> I guess that is the tricky development bit for your team, since it looks as though you use some form of Google site-specific search engine, which uses a whole whack of javascript for its implementation. I tried using a url that would do a google site specific search, which works to some extent, but which also treats doaj.org as a search term!"

Nah, that's not going to work, we don't use a Google search. We use a JS library that we wrote which queries an endpoint which is not usually recommended for use by the public. It can be seen using developer tools, but obviously use at own risk. The endpoint forwards the query to the Elasticsearch datastore, which returns the results.

Are they trying to just generate links to the DOAJ search page programmatically?

They just need a piece of code (relatively simple) to generate Elasticsearch queries, they can see the default query being sent from any search - just punch it into urldecode.org to make readable.

Even simpler, take

```
http://doaj.org/search?source=%7B%22query%22%3A%7B%22query_string%22%3A%7B%22query%22%3A%22INSERT_URLENCODED_QUERY_TEXT_HERE%22%2C%22default_operator%22%3A%22AND%22%7D%7D%2C%22from%22%3A0%2C%22size%22%3A10%7D
```

and replace the ```INSERT_URLENCODED_QUERY_TEXT_HERE``` part programmatically with whatever is needed (user input from another field, an article or journal's title, subject keywords). More complex query URLs can be observed by actually performing the search on the search page and punching it into urldecode.org, then figuring out the bit to replace, as I've done above.
